packages/lesswrong/lib/collections/posts/tableOfContents.js

```diff
@@ -160,4 +159,59 @@ function tagToHeadingLevel(tagName)
     return 0;
 }
 
+async function getTocAnswers (document) {
+  if (!document.question) return []
+
+  let answersTerms = {
+    answer:true,
+    postId: document._id,
+    deleted:false,
+  }
+  if (getSetting('forumType') === 'AlignmentForum') {
+    answersTerms.af = true
+  }
+  const answers = await Comments.find(answersTerms, {sort:questionAnswersSort}).fetch()
+  const answerSections = answers.map((answer) => ({
+    title: `${answer.baseScore} ${answer.author}`,
+    answer: answer,
+    anchor: answer._id,
+    level: 2
+  }))
+
+  if (answerSections.length) {
+    return [{anchor: "answers", level:1, title:"Answers"}, ...answerSections]
+  } else {
+    return []
+  }
+}
+
+async function getTocComments (document) {
+  const commentSelector = {
+    answer: false,
+    parentAnswerId: null,
+    postId: document._id
+  }
+  if (document.af && getSetting('forumType') === 'AlignmentForum') {
+    commentSelector.af = true
+  }
+  const commentCount = await Comments.find(commentSelector).count()
+  return [{anchor:"comments", level:0, title: Posts.getCommentCountStr(document, commentCount)}]
+}
+
+const getTableOfContentsData = async (document, args, options) => {
+  const { html } = document.contents || {}
+  let tocSections = extractTableOfContents(html) || []
+
+  tocSections.push(...(await getTocAnswers(document)))
```

Comment:
Not a huge fan of this construction. I would just get these in two earlier lines.